apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/configuration/AuthConfiguration.java

```diff
@@ -111,27 +115,42 @@ public LogoutHandler logoutHandler() {
     }
 
     @Bean
-    public static JdbcUserDetailsManager jdbcUserDetailsManager(PasswordEncoder passwordEncoder,
-        AuthenticationManagerBuilder auth, DataSource datasource) throws Exception {
+    public static JdbcUserDetailsManager jdbcUserDetailsManager(
+            PasswordEncoder passwordEncoder,
+            AuthenticationManagerBuilder auth,
+            DataSource datasource,
+            EntityManagerFactory entityManagerFactory) throws Exception {
+      char openQuote = '`';
+      char closeQuote = '`';
+      try {
+        SessionFactoryImplementor sessionFactory = entityManagerFactory.unwrap(
+                SessionFactoryImplementor.class);
+        Dialect dialect = sessionFactory.getJdbcServices().getDialect();
+        openQuote = dialect.openQuote();
+        closeQuote = dialect.closeQuote();
+      } catch (Throwable ex) {
+        //ignore
+      }
       JdbcUserDetailsManager jdbcUserDetailsManager = auth.jdbcAuthentication()
-          .passwordEncoder(passwordEncoder).dataSource(datasource)
-          .usersByUsernameQuery("select Username,Password,Enabled from `Users` where Username = ?")
-          .authoritiesByUsernameQuery(
-              "select Username,Authority from `Authorities` where Username = ?")
-          .getUserDetailsService();
+              .passwordEncoder(passwordEncoder).dataSource(datasource)
+              .usersByUsernameQuery("SELECT " + openQuote + "Username" + closeQuote + "," + openQuote + "Password" + closeQuote + "," + openQuote + "Enabled" + closeQuote + " FROM " + openQuote + "Users" + closeQuote + " where " + openQuote + "Username" + closeQuote + " = ?")
```

Comment:
This reduces the code readability a lot, maybe we could embed some placeholders in the SQL and replace the placeholders with openQuote and closeQuote? e.g.

```java
"select ^Username$, ^Password$, ^Enabled$ from ^Users$ where ^Username$ = ?".replace('^', openQuote).replace('$', closeQuote);
```

Reply:
it sounds like a specific trick, I can't say it's better. I think hibernate uses 'openQuote' way. We can rename the variable to openQ, closeQ to reduce the line length. I prefer no to change, but I can change if you insist.
Please let me know what you think

Reply:
Or we may extract some method to do the job? The main purpose is to improve the code readability.

Reply:
How about we use the built-in MessageFormat to format the SQL?

```java
.usersByUsernameQuery(MessageFormat.format("select {0}Username{1},{0}Password{1},{0}Enabled{1} from {0}Users{1} where {0}Username{1} = ?", openQuote, closeQuote));
```

Reply:
@nobodyiam Fixed